### PR TITLE
OSDOCS-9976 updated lists of instance types

### PR DESCRIPTION
**Pull request by xJustin**
Added new instance types and reorganized into the proper categories per the `rosa list instance-types` command. 

Version(s):

4.14+
Issue:

https://issues.redhat.com/browse/OSDOCS-9976
Link to docs preview:

https://73637--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_architecture/rosa_policy_service_definition/rosa-service-definition#rosa-sdpolicy-instance-types_rosa-service-definition
QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/rosa-sdpolicy-am-aws-compute-types.adoc [hunk 1]
```diff
@@ -379,8 +410,22 @@ Support for the GPU instance type software stack is provided by AWS. Ensure that
...
+- m5ad.24xlarge (96 vCPU, 384 GiB)
+- m5d.xlarge (4 vCPU, 16 GiB)
+- m5d.2xlarge (8 vCPU, 32 GiB)
+- m5d.4xlarge (16 vCPU, 64 GiB)
+- m5d.8xlarge (32 vCPU, 28 GiB)
+- m5d.12xlarge (48 vCPU, 192 GiB)
+- m5d.16xlarge (64 vCPU, 256 GiB)
+- m5d.24xlarge (96 vCPU, 384 GiB)
 
 &#8224; This instance type provides 72 logical processors on 36 physical cores.
```
Comment (EricPonvelle) [inline modules/rosa-sdpolicy-am-aws-compute-types.adoc:430]: You changed the language in the general section from provide to offer. You may want to match throughout.

File: modules/rosa-sdpolicy-am-aws-compute-types.adoc [hunk 2]
```diff
@@ -251,7 +255,13 @@
...
-- r7iz.metal-32xl (128 vCPU, 1024 GiB)
+- r7iz.metal-32xl (128 vCPU, 1,024 GiB)
+- z1d.metal (48&#135; vCPU, 384 GiB)
+- z1d.xlarge (4 vCPU, 32 GiB)
+- z1d.2xlarge (8 vCPU, 64 GiB)
+- z1d.3xlarge (12 vCPU, 96 GiB)
+- z1d.6xlarge (24 vCPU, 192 GiB)
+- z1d.12xlarge (48 vCPU, 384 GiB)
 
 &#8224; These instance types provide 96 logical processors on 48 physical cores. They run on single servers with two physical Intel sockets.
```
Comment (EricPonvelle) [inline modules/rosa-sdpolicy-am-aws-compute-types.adoc:266]: You changed the language in the general section from provide to offer. You may want to match throughout.

**Review by EricPonvelle — COMMENTED** [review]
/lgtm

File: modules/rosa-sdpolicy-am-aws-compute-types.adoc [hunk 3]
```diff
@@ -354,11 +365,31 @@ Support for the GPU instance type software stack is provided by AWS. Ensure that
 - c6id.16xlarge (64 vCPU, 128 GiB)
 - c6id.24xlarge (96 vCPU, 192 GiB)
 - c6id.32xlarge (128 vCPU, 256 GiB)
```
Comment (AkashKanni) [inline modules/rosa-sdpolicy-am-aws-compute-types.adoc:367]: The instance-types from this line to line 380 (m5zn.xlarge) are listed on same line, 
may i know if those are kept in one line intentionally, if not i think we should use a new line for each type to make it easy to read.

Comment (xJustin) [reply modules/rosa-sdpolicy-am-aws-compute-types.adoc:367]: Good catch @AkashKanni! This was not intended. Should be fixed now.

**Review by AkashKanni — APPROVED** [review]
LGTM

**Review by EricPonvelle — COMMENTED** [review]
/lgtm

**Pull request merged**